### ISSUE TITLE
Random Values identical for Multiple Instances of a Plugin

### ISSUE THREAD

**Issue by diemildefreude** [enzienaudio/heavy#203]
I have a plugin which continuously generates 'random' values for the start of a sound and its length, but when I have multiple instances of the plugin in the scene, they all seem to have the same values. Is this normal? If so, I suppose the workaround is to generate the random values in Unity and not in the patch itself.

**Comment by diplojocus** [comment]
Yes because the [random] object is deterministic, with the same seed you'll always get the same sequence of events. You can seed the object via it's inlet, possibly sending some random seed value from Unity per instance if you want.

**Comment by diemildefreude** [comment]
Does feeding a slightly different value into the right inlet of [random] for each instance solve this completely? What is it that determines the seed? The computer's clock? If so, would it suffice instead to stagger the activation of the plugin scripts?

**Comment by diplojocus** [comment]
@diemildefreude the right inlet of `[random]` doesn't set the seed in Pd. You have do something like:
```
[seed 14(
|
[random]
```

And it's down the user to make sure the random object is seeded correctly, nothing happens by default, so staggering the activation won't do anything.

One way would be to create a seed parameter to the patch that will internally set all random objects. 

Then in Unity assign each instance of that patch a different seed value, for example by just incrementing a number, or you could use some combination of the current computer/date time if you want it different on each play. If you want it _really_ random then some techniques include sampling something a bit more chaotic, like the input microphone.

**Comment by diemildefreude** [comment]
@diplojocus Ah, I could have found that out by opening [random]'s help file. Sorry! Those are some good ideas, thanks a lot.
